File: frontend/services/helpCenter.ts
import { api } from "@/lib/api";

export interface HelpCenterMessage {
  id: string;
  user_id: string;
  question: string;
  answer: string;
  category: string;
  created_at: string;
}

export interface HelpCenterQuestion {
  question: string;
}

export async function askHelpCenter(
  question: string
): Promise<HelpCenterMessage> {
  const res = await api.post(
    "/help-center/chat",
    {
      question,
    }
  );

  return res.data;
}

export async function getHelpCenterHistory(): Promise<
  HelpCenterMessage[]
> {
  const res = await api.get(
    "/help-center/history"
  );

  return res.data;
}

export async function clearHelpCenterHistory() {
  const res = await api.delete(
    "/help-center/history"
  );

  return res.data;
}

File: backend/app/schemas/help_center.py
from uuid import UUID
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class HelpCenterQuestion(BaseModel):
    question: str


class HelpCenterResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: UUID
    user_id: UUID
    question: str
    answer: str
    category: str
    created_at: datetime


class HelpCenterChatResponse(BaseModel):
    id: UUID
    question: str
    answer: str
    category: str
    created_at: datetime